### PR TITLE
Added support for fetching data with protocol buffers

### PR DESCRIPTION
**Pull request by Raalsky**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: dev_requirements.txt [hunk 1]
```diff
@@ -16,13 +16,15 @@ pytest-xdist
 vega_datasets
 backoff
 altair
+icecream
```
Comment (AleksanderWWW) [inline dev_requirements.txt:19]: Why? :smile:

Comment (AleksanderWWW) [reply dev_requirements.txt:19]: Don't get me wrong, I love this tool - I just don't see it installed anywhere

File: src/neptune/api/searching_entries.py [hunk 2]
```diff
@@ -97,7 +98,8 @@ def get_single_page(
     types: Optional[Iterable[str]],
     query: Optional["NQLQuery"],
     searching_after: Optional[str],
-) -> Any:
+    use_proto: Optional[bool] = None,
```
Comment (AleksanderWWW) [inline src/neptune/api/searching_entries.py:101]: Why is this optional? Cannot be just `use_proto: bool = False`?

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 3]
```diff
@@ -1062,7 +1069,9 @@ def search_leaderboard_entries(
         ascending: bool = False,
         progress_bar: Optional[ProgressBarType] = None,
         step_size: Optional[int] = None,
+        use_proto: Optional[bool] = None,
```
Comment (AleksanderWWW) [inline src/neptune/internal/backends/hosted_neptune_backend.py:1072]: Same here, maybe just `bool`?

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 4]
```diff
@@ -1118,6 +1128,61 @@ def get_model_version_url(
         base_url = self.get_display_address()
         return f"{base_url}/{workspace}/{project_name}/m/{model_id}/v/{sys_id}"
 
+    def get_fields_definitions(
+        self,
+        container_id: str,
+        container_type: ContainerType,
+        use_proto: Optional[bool] = None,
```
Comment (AleksanderWWW) [inline src/neptune/internal/backends/hosted_neptune_backend.py:1135]: And here (just `bool`)

File: src/neptune/api/searching_entries.py (hunk 2 above)
Comment (Raalsky) [reply src/neptune/api/searching_entries.py:101]: `None` means -> keep what environment variable says. `True`/`False` will override it. Used for testing purposes.

**Review by AleksanderWWW — APPROVED** [review]

**Pull request merged**